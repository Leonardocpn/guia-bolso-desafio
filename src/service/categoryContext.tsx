import React, { createContext, useContext, useState } from "react";

type CategoryContextType = {
  category: string;
  setCategory: (value: string) => void;
};

const CategoryContext = createContext<CategoryContextType | undefined>(
  undefined
);

export const useCategory = () => useContext(CategoryContext);

type Props = {
  children: React.ReactNode;
};
export const CategoryProvider = ({ children }: Props) => {
  const [category, setCategory] = useState("");

  return (
    <CategoryContext.Provider value={{ category, setCategory }}>
      {children}
    </CategoryContext.Provider>
  );
};
