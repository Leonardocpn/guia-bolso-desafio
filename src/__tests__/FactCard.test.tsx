import React from "react";
import { mount } from "enzyme";
import FactCard from "../pages/home/components/facts";
import { CategoryProvider } from "../service/categoryContext";

it("should render Fact Cardw", () => {
  const component = mount(
    <CategoryProvider>
      <FactCard />
    </CategoryProvider>
  );
  expect(component).toMatchSnapshot();
});
