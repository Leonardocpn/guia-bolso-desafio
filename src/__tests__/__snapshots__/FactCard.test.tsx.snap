// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Fact Cardw 1`] = `
<Component>
  <FactCard>
    <styled.div>
      <div
        className="sc-AxjAm zMyZa"
      >
        <Styled(WithStyles(ForwardRef(Card)))>
          <WithStyles(ForwardRef(Card))
            className="sc-AxiKw fENEOw"
          >
            <ForwardRef(Card)
              className="sc-AxiKw fENEOw"
              classes={
                Object {
                  "root": "MuiCard-root",
                }
              }
            >
              <WithStyles(ForwardRef(Paper))
                className="MuiCard-root sc-AxiKw fENEOw"
                elevation={1}
              >
                <ForwardRef(Paper)
                  className="MuiCard-root sc-AxiKw fENEOw"
                  classes={
                    Object {
                      "elevation0": "MuiPaper-elevation0",
                      "elevation1": "MuiPaper-elevation1",
                      "elevation10": "MuiPaper-elevation10",
                      "elevation11": "MuiPaper-elevation11",
                      "elevation12": "MuiPaper-elevation12",
                      "elevation13": "MuiPaper-elevation13",
                      "elevation14": "MuiPaper-elevation14",
                      "elevation15": "MuiPaper-elevation15",
                      "elevation16": "MuiPaper-elevation16",
                      "elevation17": "MuiPaper-elevation17",
                      "elevation18": "MuiPaper-elevation18",
                      "elevation19": "MuiPaper-elevation19",
                      "elevation2": "MuiPaper-elevation2",
                      "elevation20": "MuiPaper-elevation20",
                      "elevation21": "MuiPaper-elevation21",
                      "elevation22": "MuiPaper-elevation22",
                      "elevation23": "MuiPaper-elevation23",
                      "elevation24": "MuiPaper-elevation24",
                      "elevation3": "MuiPaper-elevation3",
                      "elevation4": "MuiPaper-elevation4",
                      "elevation5": "MuiPaper-elevation5",
                      "elevation6": "MuiPaper-elevation6",
                      "elevation7": "MuiPaper-elevation7",
                      "elevation8": "MuiPaper-elevation8",
                      "elevation9": "MuiPaper-elevation9",
                      "outlined": "MuiPaper-outlined",
                      "root": "MuiPaper-root",
                      "rounded": "MuiPaper-rounded",
                    }
                  }
                  elevation={1}
                >
                  <div
                    className="MuiPaper-root MuiCard-root sc-AxiKw fENEOw MuiPaper-elevation1 MuiPaper-rounded"
                  >
                    <styled.div>
                      <div
                        className="sc-AxirZ iiiwSz"
                      />
                    </styled.div>
                  </div>
                </ForwardRef(Paper)>
              </WithStyles(ForwardRef(Paper))>
            </ForwardRef(Card)>
          </WithStyles(ForwardRef(Card))>
        </Styled(WithStyles(ForwardRef(Card)))>
        <WithStyles(ForwardRef(Button))
          color="primary"
          onClick={[Function]}
          variant="outlined"
        >
          <ForwardRef(Button)
            classes={
              Object {
                "colorInherit": "MuiButton-colorInherit",
                "contained": "MuiButton-contained",
                "containedPrimary": "MuiButton-containedPrimary",
                "containedSecondary": "MuiButton-containedSecondary",
                "containedSizeLarge": "MuiButton-containedSizeLarge",
                "containedSizeSmall": "MuiButton-containedSizeSmall",
                "disableElevation": "MuiButton-disableElevation",
                "disabled": "Mui-disabled",
                "endIcon": "MuiButton-endIcon",
                "focusVisible": "Mui-focusVisible",
                "fullWidth": "MuiButton-fullWidth",
                "iconSizeLarge": "MuiButton-iconSizeLarge",
                "iconSizeMedium": "MuiButton-iconSizeMedium",
                "iconSizeSmall": "MuiButton-iconSizeSmall",
                "label": "MuiButton-label",
                "outlined": "MuiButton-outlined",
                "outlinedPrimary": "MuiButton-outlinedPrimary",
                "outlinedSecondary": "MuiButton-outlinedSecondary",
                "outlinedSizeLarge": "MuiButton-outlinedSizeLarge",
                "outlinedSizeSmall": "MuiButton-outlinedSizeSmall",
                "root": "MuiButton-root",
                "sizeLarge": "MuiButton-sizeLarge",
                "sizeSmall": "MuiButton-sizeSmall",
                "startIcon": "MuiButton-startIcon",
                "text": "MuiButton-text",
                "textPrimary": "MuiButton-textPrimary",
                "textSecondary": "MuiButton-textSecondary",
                "textSizeLarge": "MuiButton-textSizeLarge",
                "textSizeSmall": "MuiButton-textSizeSmall",
              }
            }
            color="primary"
            onClick={[Function]}
            variant="outlined"
          >
            <WithStyles(ForwardRef(ButtonBase))
              className="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary"
              component="button"
              disabled={false}
              focusRipple={true}
              focusVisibleClassName="Mui-focusVisible"
              onClick={[Function]}
              type="button"
            >
              <ForwardRef(ButtonBase)
                className="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary"
                classes={
                  Object {
                    "disabled": "Mui-disabled",
                    "focusVisible": "Mui-focusVisible",
                    "root": "MuiButtonBase-root",
                  }
                }
                component="button"
                disabled={false}
                focusRipple={true}
                focusVisibleClassName="Mui-focusVisible"
                onClick={[Function]}
                type="button"
              >
                <button
                  className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={0}
                  type="button"
                >
                  <span
                    className="MuiButton-label"
                  >
                    Get a new Fact
                  </span>
                  <WithStyles(memo)
                    center={false}
                  >
                    <ForwardRef(TouchRipple)
                      center={false}
                      classes={
                        Object {
                          "child": "MuiTouchRipple-child",
                          "childLeaving": "MuiTouchRipple-childLeaving",
                          "childPulsate": "MuiTouchRipple-childPulsate",
                          "ripple": "MuiTouchRipple-ripple",
                          "ripplePulsate": "MuiTouchRipple-ripplePulsate",
                          "rippleVisible": "MuiTouchRipple-rippleVisible",
                          "root": "MuiTouchRipple-root",
                        }
                      }
                    >
                      <span
                        className="MuiTouchRipple-root"
                      >
                        <TransitionGroup
                          childFactory={[Function]}
                          component={null}
                          exit={true}
                        />
                      </span>
                    </ForwardRef(TouchRipple)>
                  </WithStyles(memo)>
                </button>
              </ForwardRef(ButtonBase)>
            </WithStyles(ForwardRef(ButtonBase))>
          </ForwardRef(Button)>
        </WithStyles(ForwardRef(Button))>
      </div>
    </styled.div>
  </FactCard>
</Component>
`;
